features/device-config/monitor/src/main/java/org/opennms/features/deviceconfig/monitors/DeviceConfigMonitor.java

```diff
@@ -50,22 +59,66 @@
     public static final String PORT = "port";
     public static final String TIMEOUT = "timeout";
     public static final String PASSWORD = "password";
+    public static final String SCHEDULE = "schedule";
+    public static final String DEFAULT_CRON_SCHEDULE = "0 0 0 * * ?";
+    public static final String LAST_RETRIEVAL = "lastRetrieval";
+    public static final String CONFIG_TYPE = "config-type";
 
     private static final Logger LOG = LoggerFactory.getLogger(DeviceConfigMonitor.class);
     private Retriever retriever;
     private static final Duration DEFAULT_DURATION = Duration.ofMinutes(1); // 60sec
     private static final int DEFAULT_SSH_PORT = 22;
 
+    private IpInterfaceDao ipInterfaceDao;
+    private DeviceConfigDao deviceConfigDao;
+
+    @Override
+    public Map<String, Object> getRuntimeAttributes(final MonitoredService svc, final Map<String, Object> parameters) {
+        if (ipInterfaceDao == null) {
+            ipInterfaceDao = BeanUtils.getBean("daoContext", "ipInterfaceDao", IpInterfaceDao.class);
+        }
+
+        if (deviceConfigDao == null) {
+            deviceConfigDao = BeanUtils.getBean("daoContext", "deviceConfigDao", DeviceConfigDao.class);
+        }
+
+        final Map<String, Object> params = super.getRuntimeAttributes(svc, parameters);
+        final String configType = getKeyedString(parameters, CONFIG_TYPE, ConfigType.Default);
+        final OnmsIpInterface ipInterface = ipInterfaceDao.findByNodeIdAndIpAddress(svc.getNodeId(), svc.getIpAddr());
+        final Optional<org.opennms.features.deviceconfig.persistence.api.DeviceConfig> deviceConfigOptional = deviceConfigDao.getLatestConfigForInterface(ipInterface, configType);
+
+        if (deviceConfigOptional.isPresent()) {
+            params.put(LAST_RETRIEVAL, deviceConfigOptional.get().getLastUpdated().getTime());
```

Comment:
Since the value is not a String, parameter saves it as content but Monitor retrieve it as value.
```suggestion
            params.put(LAST_RETRIEVAL, String.valueOf(deviceConfigOptional.get().getLastUpdated().getTime()));
```

Reply:
done.